book/src/pil/types.md

```diff
@@ -0,0 +1,85 @@
+# Types
+
+The powdr-pil language has the following types:
+
+- `bool`
+- `int` (integer)
+- `fe` (field element)
+- `string`
+- tuple
+- array
+- function type
+- expression
+- identity
+
+The typing is a mix of compile-time and run-time typing. Columns are expected
+to have type `int -> fe` (i.e. a function mapping integers (row indices) to field elements).
+This is not checked, though, but if a column is evaluated as a function call
+(to retrieve a specific value in a specific row of the column), the signature `int -> fe`
+is assumed.
```

Comment:
This part can be a bit confusing. What's the "runtime typing"? Isn't it rather that columns **do have** `int -> fe` type and the math will work accordingly?